UnitTests/CardTests/CardClient_Tests.swift

```diff
@@ -34,227 +32,179 @@ class CardClient_Tests: XCTestCase {
         cardClient = CardClient(
             config: config,
-            apiClient: apiClient,
+            apiClient: mockAPIClient,
             webAuthenticationSession: mockWebAuthSession
         )
     }
 
-    override func tearDown() {
-        super.tearDown()
-        mockURLSession.clear()
-    }
-
     // MARK: - approveOrder() tests
 
     func testApproveOrder_withNoThreeDSecure_returnsOrderData() {
-        if let jsonResponse = CardResponses.confirmPaymentSourceJson.rawValue.data(using: String.Encoding.utf8) {
-            let mockResponse = MockResponse.success(
-                MockResponse.Success(data: jsonResponse, urlResponse: successURLResponse)
-            )
-            mockURLSession.addResponse(mockResponse)
-            let expectation = expectation(description: "testName")
+        mockAPIClient.cannedJSONResponse = CardResponses.confirmPaymentSourceJson.rawValue
+        
+        let expectation = expectation(description: "testName")
 
-            let cardRequest = CardRequest(orderID: "testOrderId", card: card)
+        let cardRequest = CardRequest(orderID: "testOrderId", card: card)
 
-            let mockCardDelegate = MockCardDelegate(success: {_, result -> Void in
-                XCTAssertEqual(result.orderID, "testOrderId")
-                XCTAssertEqual(result.paymentSource?.card.brand, "VISA")
-                XCTAssertEqual(result.paymentSource?.card.lastFourDigits, "7321")
-                XCTAssertEqual(result.paymentSource?.card.type, "CREDIT")
-                expectation.fulfill()
-            }, error: { _, _ -> Void in
-                XCTFail()
-            }, threeDSWillLaunch: { _ -> Void in
-                XCTFail()
-            })
+        let mockCardDelegate = MockCardDelegate(success: {_, result -> Void in
+            XCTAssertEqual(result.orderID, "testOrderId")
+            XCTAssertEqual(result.paymentSource?.card.brand, "VISA")
+            XCTAssertEqual(result.paymentSource?.card.lastFourDigits, "7321")
+            XCTAssertEqual(result.paymentSource?.card.type, "CREDIT")
+            expectation.fulfill()
+        }, error: { _, _ -> Void in
+            XCTFail()
+        }, threeDSWillLaunch: { _ -> Void in
+            XCTFail()
+        })
 
-            cardClient.delegate = mockCardDelegate
+        cardClient.delegate = mockCardDelegate
 
-            cardClient.approveOrder(request: cardRequest)
+        cardClient.approveOrder(request: cardRequest)
 
-            waitForExpectations(timeout: 10)
-        } else {
-            XCTFail("Data json cannot be null")
-        }
+        waitForExpectations(timeout: 10)
     }
 
     func testApproveOrder_whenApiCallFails_returnsError() {
-        if let jsonResponse = """
+        mockAPIClient.cannedJSONResponse = """
         {
             "some_unexpected_response": "something"
         }
-        """.data(using: String.Encoding.utf8) {
-            let mockResponse = MockResponse.success(
```

Comment:
Sadly the diff is hideous, but these tests are mostly the same, just intended to to the left we're force unwrapping the `data(using: String.Encoding.utf8)` now in `MockAPIClient`.